dcpy/connectors/edm/recipes.py

```diff
@@ -109,33 +113,39 @@ def fetch_sql(ds: Dataset, local_library_dir):
     return target_file_path
 
 
-def import_dataset(dataset: Dataset, *, local_library_dir=LIBRARY_DEFAULT_PATH):
+def import_dataset(
+    dataset: Dataset,
+    pg_client: postgres.PostgresClient,
+    *,
+    local_library_dir=LIBRARY_DEFAULT_PATH,
+):
     """Import a recipe to local data library folder and build engine."""
+    logger.info(
+        f"Importing {dataset.name} into {pg_client.database}.{pg_client.schema}"
+    )
     if dataset.version == "latest" or dataset.version is None:
         raise Exception(
             f"Cannot import a dataset without a resolved version: {dataset}"
         )
     sql_script_path = fetch_sql(dataset, local_library_dir)
 
-    postgres.execute_file_via_shell(postgres.BUILD_ENGINE_RAW, sql_script_path)
+    postgres.execute_file_via_shell(pg_client.engine_uri, sql_script_path)
+
+    with pg_client.engine.begin() as con:
+        con.execute(text(f"ALTER TABLE {dataset.name} SET SCHEMA {pg_client.schema};"))
 
-    with postgres.build_engine.connect() as con:
         con.execute(
             text(f"ALTER TABLE {dataset.name} ADD COLUMN data_library_version text;")
         )
-        con.commit()
```

Comment:
@damonmcc these are now taken care of by autocommit mode?

Reply:
yup!